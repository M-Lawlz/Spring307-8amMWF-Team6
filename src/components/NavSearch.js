import React from "react";
import "firebase/firestore";
import Button from "@material-ui/core/Button";
import Select from "react-select";
import { HashRouter, Link } from "react-router-dom";

export default class NavSearch extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      tours: [],
      isSearching: false,
      selectDropdowns: [],
      searchVal: "",
      retrieveTourPage: false,
      currentTour: [],
    };
  }

  handleNewData = (tour) => {
    /* this deals with all tour information */
    var dbTourArray = this.state.tours;
    var newDbTour = {
      tourId: tour.data().tourId,
      userEmail: tour.data().userEmail,
      location: tour.data().location,
      uploadDate: tour.data().uploadDate,
      videoUrl: tour.data().videoUrl,
      description: tour.data().description,
    };
    dbTourArray.push(newDbTour);

    /* this array is made specifically so it is compatible
       with the select attribute & dropdowns */
    var selectTourArray = this.state.selectDropdowns;
    var newDropdownTour = {
      label: tour.data().location,
      value: "" + tour.data().tourId,
    };
    selectTourArray.push(newDropdownTour);

    this.setState({
      tours: dbTourArray,
      isSearching: false,
      selectDropdowns: selectTourArray,
    });
  };

  componentDidMount() {
    const firebase = require("firebase");
    const db = firebase.firestore();
    const toursDb = db.collection("Tours");

    toursDb
      .get()
      .then((snapshot) => {
        snapshot.forEach((doc) => {
          this.handleNewData(doc);
        });
      })
      .catch(function (error) {
        console.log("Error getting document: ", error);
      });
  }

  handleNavSearch = (toSearch) => {
    if (!this.state.isSearching) {
      this.setState({
        isSearching: toSearch,
      });
    }
  };

  handleSearchChange = (value, { action }) => {
    if (action === "select-option") {
      // do something
      this.setState({
        searchVal: value.label,
      });
      var selected = this.state.tours.find((tur) => {
        return value.label === tur.location;
      });
      this.setState({
        currentTour: selected,
      });
    }
  };

  render() {
    const options = this.state.selectDropdowns;
    return (
      <div className="rowC">
        <div id="container" style={{ width: "200px" }}>
          {/* TODO: Implement at most 5 dropdowns & figure out text color */}
          <Select
            placeholder="Search..."
            options={options}
            maxMenuHeight={190}
            onClick={this.handleNavSearch}
            onChange={this.handleSearchChange}
            theme={(theme) => ({
              ...theme,
              borderRadius: 10,
              colors: {
                primary25: "lightgreen",
                primary: "teal",
                text: "black",
              },
            })}
          />
        </div>
        <HashRouter>
          <Link
            style={{ color: "inherit" }}
            to={{
              pathname: `/TourPage/${this.state.currentTour.tourId}`,
              state: this.state.currentTour,
            }}
          >
            <Button style={{ flex: 1 }}>GO</Button>
          </Link>
        </HashRouter>
      </div>
    );
  }
}
