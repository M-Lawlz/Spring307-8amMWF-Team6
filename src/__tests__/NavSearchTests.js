import NavSearch from "../components/NavSearch";
import React from "react";
import { render, shallow } from "enzyme";
import toJson from "enzyme-to-json";

test("NavSearch renders successfully.", () => {
  const navSearch = render(<NavSearch />);
  expect(toJson(navSearch)).toMatchSnapshot();
});

// test("Get new selected tour.", () => {
//   const navsearch = shallow(<NavSearch />);
//   const instance = navsearch.instance();
//   instance.updateTour(null);
//   expect(instance.state.currentTour).toBe(null);
// });