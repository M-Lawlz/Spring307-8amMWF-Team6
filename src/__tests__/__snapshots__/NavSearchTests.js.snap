// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavSearch renders successfully. 1`] = `
<div
  class="rowC"
>
  <div
    id="container"
    style="width:200px"
  >
    <div
      class=" css-2b097c-container"
    >
      <div
        class=" css-1586ewl-control"
      >
        <div
          class=" css-g1d714-ValueContainer"
        >
          <div
            class=" css-dzw6ua-placeholder"
          >
            Search...
          </div>
          <div
            class="css-ybs8ec-Input"
          >
            <div
              class=""
              style="display:inline-block"
            >
              <input
                aria-autocomplete="list"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                id="react-select-2-input"
                spellcheck="false"
                style="box-sizing:content-box;width:1px;label:input;background:0;border:0;font-size:inherit;opacity:1;outline:0;padding:0;color:inherit"
                tabindex="0"
                type="text"
                value=""
              />
              <div
                style="position:absolute;top:0;left:0;visibility:hidden;height:0;overflow:scroll;white-space:pre"
              />
            </div>
          </div>
        </div>
        <div
          class=" css-1hb7zxy-IndicatorsContainer"
        >
          <span
            class=" css-bqiiy2-indicatorSeparator"
          />
          <div
            aria-hidden="true"
            class=" css-kaqmzc-indicatorContainer"
          >
            <svg
              aria-hidden="true"
              class="css-6q0nyr-Svg"
              focusable="false"
              height="20"
              viewBox="0 0 20 20"
              width="20"
            >
              <path
                d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>
  <a
    href="#/TourPage/undefined"
    style="color:inherit"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text"
      style="flex:1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        GO
      </span>
    </button>
  </a>
</div>
`;
